### PR TITLE
Improve Docker Compose section, refs #12721

### PR DESCRIPTION
**Pull request by jraddaoui**

### CODE REVIEW

File: dev-manual/env/compose.rst [hunk 1]
```diff
@@ -42,9 +44,14 @@ change your current directory.
 
 .. code-block:: bash
 
-   git clone -b stable/2.4.x https://github.com/artefactual/atom.git atom
+   git clone -b qa/2.5.x https://github.com/artefactual/atom.git atom
    cd atom
 
+.. note::
+
+   Clone the "stable/2.4.x" branch if you want to test the latest stable
```
Comment (fiver-watson) [inline dev-manual/env/compose.rst:61]: Do we want to specify an exact version here? If so, then we need to remember to review and update this with each major release. Might we suggest cloning the latest stable branch (e.g. `stable/2.4.x`, etc) or something more generic instead?

Comment (jraddaoui) [reply dev-manual/env/compose.rst:61]: Ideally that could be handled with Sphinx vars, I don't really know a lot about this but I think the AM docs use something like that to specify the latest stable and qa branches. In the time being, that's something that we already have to do one the release cycle when the new 2.6 branch is created. If I remember well it's needed in quite a few places, so one more shouldn't be a problem.

File: dev-manual/env/compose.rst [hunk 2]
```diff
@@ -9,7 +9,9 @@ are developed, built, distributed and deployed. The AtoM team is experimenting
 with new workflows that make use of containers. This document introduces our
 new development workflow based on Docker and `Docker Compose <https://docs.docker.com/compose/>`__.
 The latter is a tool that help us to run multi-container applications like AtoM
-and it is suitable for both development and production environments.
+and it is suitable for both development and production environments. However,
+the following Docker Compose integration and the AtoM Dockerfile are not fully
+prepared for production environmets at this point.
```
Comment (fiver-watson) [inline dev-manual/env/compose.rst]: typo here - "environmets" - should be "environments" ;)

I also wonder if this should be put in an admonition to draw attention to it? E.g.:

```
.. _User forum: https://groups.google.com/forum/#!forum/ica-atom-users

.. WARNING:
   
   The Docker Compose integration and AtoM Dockerfile outlined here is intended for **testing and development 
   purposes only**. It has not been fully tested and prepared for use in production environments at this time. 

   Artefactual provides this documentation as-is - we do not regularly test or maintain the instructions outlined 
   here as part of our release management process. If you encounter issues or bugs, you are welcome to report them 
  via the AtoM User forum_ - we will provide basic support as we are able, but urge you to proceed at your own risk.
```

File: dev-manual/env/compose.rst (hunk 1 above)
Comment (fiver-watson) [reply dev-manual/env/compose.rst:61]: Hmm, maybe nevermind, looks like we've been doing this already anyways

Comment (fiver-watson) [reply dev-manual/env/compose.rst:61]: Oh just saw your reply. Yes, we can look into that in the future. For now I think it's fine.

**Comment by fiver-watson** [comment]
Looks good! thanks for the changes! 👍 to merge

**Comment by jraddaoui** [comment]
Thanks @fiver-watson!

6edd958 in 2.5.

**Pull request merged**